src/display_backend.rs

```diff
@@ -1,11 +1,171 @@
 pub use platform::*;
 
-#[cfg(feature = "no-x11")]
+#[cfg(feature = "no-x11-wayland")]
 mod platform {
     use crate::config::{Side, StrutDefinition};
     use anyhow::*;
     pub fn reserve_space_for(window: &gtk::Window, monitor: gdk::Rectangle, strut_def: StrutDefinition) -> Result<()> {
-        Err(anyhow!("Cannot reserve space on non-X11 backends"))
+        Err(anyhow!("Cannot reserve space on non X11 or and wayland backends"))
+    }
+}
+
+#[cfg(feature = "wayland")]
+mod platform {
+    use crate::config::{Side, StrutDefinition};
+    use gtk::prelude::*;
+    use anyhow::*;
+
+    pub fn reserve_space_for(window: &gtk::Window, monitor: gdk::Rectangle, strut_def: StrutDefinition) -> Result<()> {
+        // Initializing the layer surface
+        let backend = WaylandBackend::new()?;
+        backend.reserve_space_for(window, monitor, strut_def)?;
+        Ok(())
+    }
+
+    struct WaylandBackend {
+        layer: gtk_layer_shell::Layer;
+        conn: RustConnection<DefaultStream>,
+        // I have no idea what this is. Might look later
+        // root_window: u32,
+    }
+
+    impl WaylandBackend {
+        fn new() -> Result<Self> {
+            let (conn, screen_num) = RustConnection::connect(None)?;
+            Ok((WaylandBackend {
+                layer,
+                conn,
+            }))
+        }
+
+        fn reserve_space_for()
+            &self,
+            window: &gtk::Window,
+            monitor_rect: gdk::Rectangle,
+            strut_def: StrutDefinition,
+        ) -> Result<()> {
+            let win_id = window
+                .get_window()
+                .context("Couldn't get gdk window from gtk window")?
+                .ok()
+                .context("Failed to get layer shell surface for gtk window")? // A modifier
+                // .get_xid() as u32;
+            let root_window_geometry = self.conn.get_geometry(self.root_window)?.reply()?;
+
+            // Initialising a layer shell surface
+            gtk_layer_shell::init_for_window(window);
+            // Set the layer where the layer shell surface will spawn
+            gtk_layer_shell::set_layer(window, self.layer);
+            // Anchoring the surface to an edge
+            self.set_anchor(strut_def, window);
+            // I don't like the way NumWithWidth is used to define margins
```

Comment:
It pretty much has to be though, as it's the only way to make it possible to set margins in percentage - which definitely should be possible

Reply:
Fair.

Reply:
Since percentage works with the output dimension, I think I'll need to use the output management protocol. Probably with [smithay](https://docs.rs/smithay-client-toolkit/0.12.2/smithay_client_toolkit/).